prototype_source/semi_structured_sparse.rst

```diff
@@ -0,0 +1,542 @@
+We will explain this process in this tutorial.
+
+.. image:: ../../_static/img/pruning_flow.jpg
+
+By the end of this tutorial, we will have sparsified a BERT question-answering model to be 2:4 sparse, fine-tuning it to recover nearly all F1 loss (86.92 dense vs 86.48 sparse).
+Finally, we will accelerate this 2:4 sparse model for inference, yielding a 1.3x speedup.
+
+Requirements
+--------------
+
+* PyTorch >= 2.1.
+* A NVIDIA GPU with semi-structured sparsity support (Compute Capability 8.0+).
+
+.. note::
+
+    This tutorial is designed for beginners to semi-structured sparsity / sparsity in general.
+    For users with existing 2:4 sparse models, accelerating ``nn.Linear`` layers for inference is as easy as running the following:
+
+    .. code:: python
+
+        import torch
+        from torch.sparse import to_sparse_semi_structured, SparseSemiStructuredTensor
+        from torch.utils.benchmark import Timer
+        SparseSemiStructuredTensor._FORCE_CUTLASS = True
+
+        # mask Linear weight to be 2:4 sparse
+        mask = torch.Tensor([0, 0, 1, 1]).tile((3072, 2560)).cuda().bool()
+        linear = torch.nn.Linear(10240, 3072).half().cuda().eval()
+        linear.weight = torch.nn.Parameter(mask * linear.weight)
+
+        x = torch.rand(3072, 10240).half().cuda()
+
+        with torch.inference_mode():
+            dense_output = linear(x)
+            dense_t = Timer(stmt="linear(x)",
+                            globals={"linear": linear,
+                                     "x": x}).blocked_autorange().median * 1e3
+
+            # acclerate via SparseSemiStructuredTensor
+            linear.weight = torch.nn.Parameter(to_sparse_semi_structured(linear.weight))
+
+            sparse_output = linear(x)
+            sparse_t = Timer(stmt="linear(x)",
+                            globals={"linear": linear,
+                                     "x": x}).blocked_autorange().median * 1e3
+
+            # sparse and dense matmul are numerically equivalent
+            assert torch.allclose(sparse_output, dense_output, atol=1e-3)
+            print(f"Dense: {dense_t:.3f}ms Sparse: {sparse_t:.3f}ms | Speedup: {(dense_t / sparse_t):.3f}x")
+
+    On my machine (A100 80GB), I see: `Dense: 0.870ms Sparse: 0.630ms | Speedup: 1.382x`
+
+
+What problem does semi-structured sparsity solve?
+-------------------------------------------------
+The general motivation behind sparsity is simple: if there are zeros in your network, you can avoid storing / doing compute with those parameters.
+However, the specifics of sparsity are tricky. Zeroing out parameters doesn't affect the latency / memory overhead of our model out of the box.
+
+This is because the dense tensor itself still contains the pruned elements and will still operate on those elements during matrix multiplication.
```

Comment:
introduce the word kernel here. 

"This is because the dense tensor itself still contains the pruned elements, and the respective matrix multiplication kernel will still operate on those elements."